test/unit/SourceLocator/Type/DirectorySourceLocatorTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: HP
+ * Date: 09/24/2016
+ * Time: 5:44 AM
+ */
+
+namespace BetterReflectionTest\SourceLocator\Type;
+
+
+use BetterReflection\Reflector\ClassReflector;
+use BetterReflection\SourceLocator\Type\DirectorySourceLocator;
+
+class DirectorySourceLocatorTest extends \PHPUnit_Framework_TestCase
+{
+
+
+    public function testUtilDirectory(){
+        $directoryToScan = __DIR__.'/../../Assets/DirectoryScannerAssets';
+        $sourceLocator = new DirectorySourceLocator([$directoryToScan]);
+        $reflector = new ClassReflector($sourceLocator);
+        $this->assertEquals(2, count($reflector->getAllClasses()));
```

Comment:
use `self::assertCount()`. Also make sure that the paths of the reflected classes correspond to the given ones.
You also want to test deeper nesting levels, and verify that non-`.php` files are ignored by the iterator.